core/app/models/spree/unit_cancel.rb

```diff
@@ -45,6 +45,9 @@ def weighted_line_item_amount(line_item)
   end
 
   def quantity_of_line_item(line_item)
-    BigDecimal(line_item.inventory_units.not_canceled.reject(&:original_return_item).size)
+    not_canceled_inventory_units_count = line_item.inventory_units.not_canceled.reject(&:original_return_item).size
+    raise "Line Item does not have any inventory units available to cancel" if not_canceled_inventory_units_count.zero?
```

Comment:
I think we should raise this error in the `weighted_line_item_amount ` method since this one is only a getter for the size and maybe, semantically, is allowed to return 0. `weighted_line_item_amount`, instead, cannot perform the operation it has been written for so I think that's the right place where to put this raise. What do you think?

Reply:
I like it. Makes sense to raise when the calculation is about to fail.